python/lsst/ip/diffim/dipoleFitTask.py

```diff
@@ -0,0 +1,921 @@
+            if self.negImage is not None:
+                negSubim = afwImage.MaskedImageF(self.negImage.getMaskedImage(), bbox, afwImage.PARENT)
+            else:  # no template provided; generate it from the obs. and diffim
+                negSubim = posSubim.clone()
+                negSubim -= subim
+            z = np.append([z], [posSubim.getArrays()[0],
+                                negSubim.getArrays()[0]], axis=0)
+            # Weight the pos/neg images by rel_weight relative to the diffim
+            weights = np.append([weights], [1. / posSubim.getArrays()[2] * rel_weight,
+                                            1. / negSubim.getArrays()[2] * rel_weight], axis=0)
+        else:
+            rel_weight = 0.  # a short-cut for "don't include the pre-subtraction data"
+
+        # Create the lmfit model (lmfit uses scipy 'leastsq' option by default - Levenberg-Marquardt)
+        gmod = lmfit.Model(DipoleFitAlgorithm.genDipoleModel, verbose=verbose)
+
+        # Add the constraints for centroids, fluxes.
+        # starting constraint - near centroid of footprint
+        fpCentroid = np.array([fp.getCentroid().getX(), fp.getCentroid().getY()])
+        cenNeg = cenPos = fpCentroid
+
+        pks = fp.getPeaks()
+        if len(pks) >= 1:
+            cenPos = pks[0].getF()    # if individual (merged) peaks were detected, use those
+        if len(pks) >= 2:
+            cenNeg = pks[1].getF()
+
+        # For close/faint dipoles the starting locs (min/max) might be way off, let's help them a bit.
+        # First assume dipole is not separated by more than 5*psfSigma.
+        maxSep = self.psfSigma * maxSepInSigma
+
+        # Note - this may be a cheat to assume the dipole is centered in center of the footprint.
+        if np.sum(np.sqrt((np.array(cenPos) - fpCentroid)**2.)) > maxSep:
+            cenPos = fpCentroid
+        if np.sum(np.sqrt((np.array(cenNeg) - fpCentroid)**2.)) > maxSep:
+            cenPos = fpCentroid
+
+        # parameter hints/constraints: https://lmfit.github.io/lmfit-py/model.html#model-param-hints-section
+        # might make sense to not use bounds -- see http://lmfit.github.io/lmfit-py/bounds.html
+        # also see this discussion -- https://github.com/scipy/scipy/issues/3129
+        gmod.set_param_hint('xcenPos', value=cenPos[0],
+                            min=cenPos[0]-maxSep, max=cenPos[0]+maxSep)
+        gmod.set_param_hint('ycenPos', value=cenPos[1],
+                            min=cenPos[1]-maxSep, max=cenPos[1]+maxSep)
+        gmod.set_param_hint('xcenNeg', value=cenNeg[0],
+                            min=cenNeg[0]-maxSep, max=cenNeg[0]+maxSep)
+        gmod.set_param_hint('ycenNeg', value=cenNeg[1],
+                            min=cenNeg[1]-maxSep, max=cenNeg[1]+maxSep)
+
+        # Estimate starting flux. This strongly affects runtime performance so we want to make it close.
+        if False:
```

Comment:
If you must leave in disabled code please explain clearly why you are disabling it and leaving it in. Take it out if you can.